roost_backend/views.py

```diff
@@ -51,13 +51,17 @@ def post(self, request):
             return Response("Roost-ng does not support multi-step GSS handshakes.", status=status.HTTP_400_BAD_REQUEST)
 
         principal = str(ctx.initiator_name)
-        if (serializer.validated_data['create_user']
-                and settings.ROOST_ALLOW_USER_CREATION
+        if serializer.validated_data['create_user']:
```

Comment:
This check without also looking at `settings.ROOST_ALLOW_USER_CREATION` changes the behavior of roost-ng unintentionally. The original roost clients always send `createUser: 1`, and when user creation is disabled we always fall through to a lookup case. With this change user authentication for such clients requires user creation be turned on or that they be in the allow list, even if they already exist. So, if user creation is disabled and the allowlist is empty, then users can't log in.